radicale/storage.py

```diff
@@ -621,43 +689,150 @@ def move(cls, item, to_collection, to_href):
+                    raise UnsafePathError(href)
+                path = path_to_filesystem(self._filesystem_path, href)
+            except ValueError:
+                self.logger.debug(
+                    "Can't translate name safely to filesystem: %s", href)
+                return None, None
+        else:
+            path = os.path.join(self._filesystem_path, href)
+        try:
+            with open(path, "rb") as f:
+                btext = f.read()
+        except (FileNotFoundError, IsADirectoryError):
+            return None, None
+        # The hash of the component in the file system. This is used to check,
+        # if the entry in the cache is still valid.
+        input_hash = md5()
+        input_hash.update(btext)
+        input_hash = input_hash.hexdigest()
+        cache_folder = os.path.join(self._filesystem_path, ".Radicale.cache",
+                                    "item")
+        try:
+            with open(os.path.join(cache_folder, href), "rb") as f:
+                cinput_hash, cetag, ctext, ctag, cstart, cend = pickle.load(f)
+        except FileNotFoundError:
+            cinput_hash = cetag = ctext = ctag = cstart = cend = None
+        vobject_item = None
+        if input_hash != cinput_hash:
+            vobject_item = Item(self, href,
+                                text=btext.decode(self.encoding)).item
+            ctext = vobject_item.serialize()
+            cetag = get_etag(ctext)
+            ctag, cstart, cend = xmlutils.find_tag_and_time_range(vobject_item)
+            self._makedirs_synced(cache_folder)
+            try:
+                # Race: Other processes might have created and locked the
+                # file.
+                with self._atomic_write(os.path.join(cache_folder, href),
+                                        "wb") as f:
+                    pickle.dump((input_hash, cetag, ctext,
+                                 ctag, cstart, cend), f)
+            except PermissionError:
+                pass
+            # Clean cache entries (max once per request)
+            if not self._item_cache_cleaned:
+                self._item_cache_cleaned = True
+
+                def is_invalid_href(href):
+                    return not os.path.isfile(
+                        os.path.join(self._filesystem_path, href))
+
+                self._clean_cache(cache_folder, filter(
+                    is_invalid_href, scandir(cache_folder)))
         last_modified = time.strftime(
             "%a, %d %b %Y %H:%M:%S GMT",
             time.gmtime(os.path.getmtime(path)))
-        try:
-            item = vobject.readOne(text)
-        except Exception:
-            self.logger.error("Failed to parse component: %s", href)
-            raise
-        return Item(self, item, href, last_modified)
+        return Item(self, href, last_modified=last_modified, etag=cetag,
+                    text=ctext, item=vobject_item), (ctag, cstart, cend)
```

Comment:
You can't return ``vobject_item`` directly. The ``text`` is the raw text from the file system. You have to serialize it again, to normalize the text with **vobject**. The user might have edited the storage in the file system by hand, it might not be normalized.